test/functional/test-resources.js

```diff
@@ -1824,6 +1833,67 @@ describe('Resources mutateElement and collapse', () => {
     });
   });
 
+  it('attemptCollapse should not call attemptChangeSize', () => {
```

Comment:
Ah, I confused the sync and async paths. I think re-ordering the calls might be best.